Database fails to startup sometimes

### What action do you want to perform

Just running my test suite

### What are the results

Every so often I get this error on one of the tests

```
.tox/py38/lib/python3.8/site-packages/pytest_postgresql/janitor.py:96: in __enter__

    self.init()

.tox/py38/lib/python3.8/site-packages/pytest_postgresql/janitor.py:55: in init

    with self.cursor() as cur:

../../../.pyenv/versions/3.8.7/lib/python3.8/contextlib.py:113: in __enter__

    return next(self.gen)

.tox/py38/lib/python3.8/site-packages/pytest_postgresql/janitor.py:80: in cursor

    conn = psycopg2.connect(

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 



dsn = "dbname=postgres user=postgres password='' host=127.0.0.1 port=21440"

connection_factory = None, cursor_factory = None

kwargs = {'dbname': 'postgres', 'host': '127.0.0.1', 'password': '', 'port': 21440, ...}

kwasync = {}



    def connect(dsn=None, connection_factory=None, cursor_factory=None, **kwargs):

        """

        Create a new database connection.

    

        The connection parameters can be specified as a string:

    

            conn = psycopg2.connect("dbname=test user=postgres password=secret")

    

        or using a set of keyword arguments:

    

            conn = psycopg2.connect(database="test", user="postgres", password="secret")

    

        Or as a mix of both. The basic connection parameters are:

    

        - *dbname*: the database name

        - *database*: the database name (only as keyword argument)

        - *user*: user name used to authenticate

        - *password*: password used to authenticate

        - *host*: database host address (defaults to UNIX socket if not provided)

        - *port*: connection port number (defaults to 5432 if not provided)

    

        Using the *connection_factory* parameter a different class or connections

        factory can be specified. It should be a callable object taking a dsn

        argument.

    

        Using the *cursor_factory* parameter, a new default cursor factory will be

        used by cursor().

    

        Using *async*=True an asynchronous connection will be created. *async_* is

        a valid alias (for Python versions where ``async`` is a keyword).

    

        Any other keyword parameter will be passed to the underlying client

        library: the list of supported parameters depends on the library version.

    

        """

        kwasync = {}

        if 'async' in kwargs:

            kwasync['async'] = kwargs.pop('async')

        if 'async_' in kwargs:

            kwasync['async_'] = kwargs.pop('async_')

    

        if dsn is None and not kwargs:

            raise TypeError('missing dsn and no parameters')

    

        dsn = _ext.make_dsn(dsn, **kwargs)

>       conn = _connect(dsn, connection_factory=connection_factory, **kwasync)

E       psycopg2.OperationalError: FATAL:  the database system is starting up



.tox/py38/lib/python3.8/site-packages/psycopg2/__init__.py:127: OperationalError
```

### What are the expected results

Whatever mechanism waits for the db to be responsive needs to be more robust, such that no tests start until the db is fully up.

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -17,6 +17,9 @@ unreleased
 - [development] Fully moved to the Github Actions as CI/CD pipeline
 - [bugfix] Use `postgresql_logsprefix` and `--postgresql-logsprefix` again.
   They weren't used somewhere along the way.
+- [bugfix] Sometimes pytest-postrgesql would fail to start postgresql with
+  "FATAL:  the database system is starting up" message. It's not really a fatal error,
+  but a message indicating that the process still starts. Now pytest-postgresql will wait properly in this cases.
 
 2.6.1
 ----------

